CRM/Core/BAO/Setting.php

```diff
@@ -519,19 +513,8 @@ public static function isAPIJobAllowedToRun($params): void {
    */
   public static function onChangeEnvironmentSetting($oldValue, $newValue, $metadata) {
     if ($newValue !== 'Production') {
-      $mailing = Civi::settings()->get('mailing_backend');
-      if ($mailing['outBound_option'] != CRM_Mailing_Config::OUTBOUND_OPTION_DISABLED) {
-        Civi::settings()->set('mailing_backend_store', $mailing);
-      }
-      Civi::settings()->set('mailing_backend', ['outBound_option' => CRM_Mailing_Config::OUTBOUND_OPTION_DISABLED]);
```

Comment:
I'm not sure I agree with removing this specific line, or the whole test. The intent of this code seemed to be to make sure you didn't accidentally send out mails when not in production. Perhaps @monishdeb or @JoeMurray can confirm.

I do agree with the part stated in the PR description, that running a job while not in production shouldn't magically set the outbound mail back to some potentially random value, that if you want mail to go out while doing that special run you can manually update the setting.

Reply:
Good point - I'd be OK with keeping that specific line (and the status message, and part of the test). At least that part is fairly clear to admins.

Reply:
Another option is get rid of both the environment setting and runInNonProduction, since it can be replaced by what a setup script would do and most consultants already have such scripts, and "normal" people seem unlikely to use this setting. But that's a larger discussion.

Reply:
The problem I have with line 526 is that assumes that I want to completely disable mailing. But we want to instead have the mailings captured by a different app. So long as we can still set `$civicrm_setting['domain']['mailing_backend']['outBound_option'] = 3;` regardless of what the environment is set to.

Reply:
Although that would be a bigger change because then anyone who has a script that sets environment might be expecting it to automatically disable outbound mail and it no longer would, and they wouldn't see the status message if it's done via script.